src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -34,8 +37,8 @@ public static bool RequiresReflectionMethodBodyScannerForCallSite(FlowAnnotation
             return
                 GetIntrinsicIdForMethod(methodDefinition) > IntrinsicId.RequiresReflectionBodyScanner_Sentinel ||
                 flowAnnotations.RequiresDataflowAnalysis(methodDefinition) ||
-                methodDefinition.HasCustomAttribute("System.Diagnostics.CodeAnalysis", "RequiresUnreferencedCodeAttribute") ||
-                methodDefinition.HasCustomAttribute("System.Diagnostics.CodeAnalysis", "RequiresDynamicCodeAttribute") ||
+                methodDefinition.DoesMethodRequires("RequiresUnreferencedCodeAttribute", out _) ||
+                methodDefinition.DoesMethodRequires("RequiresDynamicCodeAttribute", out _) ||
```

Comment:
Please define the names of the attributes as constants somewhere and use those everywhere.